File: apps/web/src/types/consent.ts
// ============================================================
// consent.ts — Consentimiento informado
//
// Backend: ConsentsController (/api/v1/consent-templates, /api/v1/consents).
// Variables de plantilla: {{paciente}} {{cedula}} {{procedimiento}} {{fecha}}.
// ============================================================

export const CONSENT_VARIABLES = ["{{paciente}}", "{{cedula}}", "{{procedimiento}}", "{{fecha}}"] as const;

export interface ConsentTemplate {
  procedureId: string;
  procedimiento: string;
  requiereConsentimiento: boolean;
  titulo: string;
  texto: string;
  /** true: el tenant aún no la editó (se muestra la plantilla del sistema). */
  esPorDefecto: boolean;
  updatedAt: string | null;
}

export interface ConsentPreview {
  titulo: string;
  texto: string;
  paciente: string;
  cedula: string;
  procedimiento: string;
}

export interface SignConsentRequest {
  patientId: string;
  procedureId: string;
  appointmentId?: string;
  /** data URL PNG de signature_pad */
  firmaPng: string;
}

export interface Consent {
  id: string;
  patientId: string;
  procedureId: string;
  procedimiento: string;
  appointmentId: string | null;
  esteticista: string;
  fechaFirma: string;
  titulo: string;
  /** PDF firmado (se abre con su URL firmada). */
  attachmentId: string;
}

File: apps/api/src/NemediClinic.Domain/Entities/Consent.cs
namespace NemediClinic.Domain.Entities;

/// <summary>
/// Plantilla de consentimiento informado de un procedimiento, editable por cada tenant.
/// Variables: {{paciente}}, {{cedula}}, {{procedimiento}}, {{fecha}}.
/// </summary>
public class ConsentTemplate : BaseEntity
{
    public Guid ProcedureId { get; set; }
    public string Titulo { get; set; } = string.Empty;
    public string Texto { get; set; } = string.Empty;

    public Procedure Procedure { get; set; } = null!;
}

/// <summary>
/// Consentimiento firmado. TextoFirmado es la foto del texto ya con las variables resueltas: lo que
/// la persona leyó y firmó, aunque después cambie la plantilla. El PDF es un Attachment
/// (EntityType = Patient, Kind = Consentimiento).
/// </summary>
public class Consent : BaseEntity
{
    public Guid PatientId { get; set; }
    public Guid ProcedureId { get; set; }
    public Guid? AppointmentId { get; set; }
    public Guid EsteticistId { get; set; }
    public DateTime FechaFirma { get; set; } = DateTime.Now;
    public string Titulo { get; set; } = string.Empty;
    public string TextoFirmado { get; set; } = string.Empty;
    public string FirmanteNombre { get; set; } = string.Empty;
    public string FirmanteCedula { get; set; } = string.Empty;
    public Guid AttachmentId { get; set; }

    public Patient Patient { get; set; } = null!;
    public Procedure Procedure { get; set; } = null!;
    public User Esteticist { get; set; } = null!;
}
